web/src/client/network.test.js

```diff
@@ -0,0 +1,278 @@
+ *
+ * To contact SUSE LLC about this file by physical or electronic mail, you may
+ * find current contact information at www.suse.com.
+ */
+
+// @ts-check
+
+import {
+  CONNECTION_STATE, CONNECTION_TYPES, NetworkClient, formatIp, NetworkManagerAdapter
+} from "./network";
+import { DBusClient } from "./dbus";
+
+const NM_IFACE = "org.freedesktop.NetworkManager";
+const NM_SETTINGS_IFACE = "org.freedesktop.NetworkManager.Settings";
+const NM_ACTIVE_CONNECTION_IFACE = "org.freedesktop.NetworkManager.Connection.Active";
+const NM_CONNECTION_IFACE = "org.freedesktop.NetworkManager.Settings.Connection";
+const NM_IP4CONFIG_IFACE = "org.freedesktop.NetworkManager.IP4Config";
+
+const dbusClient = new DBusClient("");
+const activeConnections = {
+  "/active/connection/wifi/1": {
+    Id: "active-wifi-connection",
+    path: "/active/wifi/connection",
+    Connection: "/active/connection/wifi/1",
+    Devices: ["hardware/wifi/1"],
+    State: CONNECTION_STATE.ACTIVATED,
+    Type: CONNECTION_TYPES.WIFI,
+    Ip4Config: "/ip4Config/2"
+  },
+  "/active/connection/wired/1": {
+    Id: "active-wired-connection",
+    path: "/active/wired/connection",
+    Connection: "/active/connection/wired/1",
+    Devices: ["hardware/wired/1"],
+    State: CONNECTION_STATE.ACTIVATED,
+    Type: CONNECTION_TYPES.ETHERNET,
+    Ip4Config: "/ip4Config/1"
+  }
+};
+
+const ipv4SettingsMock = {
+  "address-data": {
+    t: "aa{sv}",
+    v: [
+      {
+        address: {
+          t: "s",
+          v: "192.168.68.254"
+        },
+        prefix: {
+          t: "u",
+          v: 24
+        }
+      }
+    ]
+  },
+  addresses: {
+    t: "aau",
+    v: [
+      [
+        4265912512,
+        24,
+        21276864
+      ]
+    ]
+  },
+  "dns-search": {
+    t: "as",
+    v: []
+  },
+  gateway: {
+    t: "s",
+    v: "192.168.68.1"
+  },
+  method: {
+    t: "s",
+    v: "manual"
+  },
+  "route-data": {
+    t: "aa{sv}",
+    v: []
+  },
+  routes: {
+    t: "aau",
+    v: []
+  }
+};
+
+const connectionSettingsMock = {
+  Update: jest.fn(),
+  GetSettings: () => ({ ipv4: ipv4SettingsMock }),
+};
+
+const networkProxy = {
+  wait: jest.fn(),
+  ActivateConnection: jest.fn(),
+  ActiveConnections: Object.keys(activeConnections)
+};
+
+const networkSettingsProxy = {
+  wait: jest.fn(),
+  Hostname: "testing-machine"
+};
+
+const activeConnectionProxy = path => {
+  return activeConnections[path];
+};
+
+// TODO: return a mock for each connection path
+const connectionProxy = path => connectionSettingsMock;
+
+const addressesData = {
+  "/ip4Config/1": {
+    wait: jest.fn(),
+    AddressData: [
+      {
+        address: { v: "10.0.0.1", t: "s" },
+        prefix: { v: "22", t: "s" }
+      }
+    ]
+  },
+  "/ip4Config/2": {
+    wait: jest.fn(),
+    AddressData: [
+      {
+        address: { v: "10.0.0.2", t: "s" },
+        prefix: { v: "22", t: "s" }
+      }
+    ]
+  }
+};
+
+const ipConfigProxy = path => {
+  return addressesData[path];
+};
+
```

Comment:
We are testing a bare minimum here, reason why there is a lot of room for improvements in this unit test.